spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpSendingMessageHandler.java

```diff
@@ -110,7 +113,10 @@ public void handleMessageInternal(final Message<?> message) throws
 			}
 			else {
 				logger.error("Unable to find outbound socket for " + message);
-				throw new MessageHandlingException(message, "Unable to find outbound socket");
+				MessageHandlingException messageHandlingException = new MessageHandlingException(message,
+						"Unable to find outbound socket");
+				publishNoConnectionEvent(message, (String) connectionId);
```

Comment:
I wonder if we can propagate this `messageHandlingException` to the even instead of that new one.